### PR TITLE
Remove `anyhow` from trait definitions

### LINKED ISSUE #80: Remove `anyhow` dependency
I'm trying to use `eyre` with this crate and getting some really gnarly generic trait bound errors.  It appears it's because functions like `receive_activity` and the like are coupled to ensuring that something can implement the `anyhow::Error` type.

I propose we should refactor this out, and rely solely on `thiserror` for errors within this crate and provide some way of bringing your own errors for library consumers.

Let me know if you're open to this and I can have a crack

### PR DESCRIPTION
**Pull request by cetra3**
Removes `anyhow` dependency in the activitypub crate.  Resolves #80 but is a semver breaking change.

I've tested this works fine with *minor* changes to the main `lemmy` crate on this branch here: https://github.com/LemmyNet/lemmy/compare/main...cetra3:lemmy:remove_anyhow

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -27,17 +33,35 @@ pub enum Error {
...
+    /// Url Parse
+    #[error(transparent)]
+    UrlParse(#[from] url::ParseError),
+    /// Signing errors
     #[error(transparent)]
-    Other(#[from] anyhow::Error),
+    SignError(#[from] SignError),
+    /// Other generic errors
+    #[error("{0}")]
+    Other(String),
```
Comment (Nutomic) [inline src/error.rs:59]: Much better with separate errors :+1:

**Review by Nutomic — APPROVED** [review]
The Lemmy code changes are minimal so this looks very reasonable. Will give @phiresky and @dessalines a chance to review before merging.

**Review by dessalines — APPROVED** [review]

**Comment by Nutomic** [comment]
Thanks, I merged the PR and published `0.5.0-beta.5`. Can you open a PR for the Lemmy changes?

**Comment by Nutomic** [comment]
Thanks! I published 0.5.0-beta.5 with this, can you open a PR for Lemmy?

**Pull request merged**